src/ui/src/app/shared/services/charts.service.ts

```diff
@@ -127,6 +127,19 @@ export class ChartsService {
       .catch(this.handleError);
   }
 
+  /**
+   * Get the URL for retrieving the chart's icon
+   * 
+   * @param {string} chart Chart object
+   */
+  getChartIconURL(chart: Chart): string {
+    if (chart.attributes.icon) {
+      return `${this.hostname}${chart.attributes.icon}`
+    } else {
+      return null;
```

Comment:
Maybe you can return here the placeholder so the caller does not do the check also?

Reply:
Which seems to happen in multiple places.

Reply:
I was thinking about it, but the placeholder is not something from the chartsvc, but part of the Angular app, so it might be a bit weird. However, it would reduce boilerplate so maybe it's ok to put it here.

Reply:
The chartSVC is a consumer class in the frontend no? Not something that we import/re-use from the chart-svc? In which case I think it's ok to move the placeholder here. 

I'd understand that we wouldn't want to do it if we had different placeholders. Up to you though.

Reply:
Yes it is, but I figured it should keep the context of talking to the chartsvc, but I agree it doesn't make much sense to have the if/else logic repeated everywhere we want to use it. Will move it here then. Thanks!